pkg/pq/priority_queue.go

```diff
@@ -51,16 +51,31 @@ func (pq *PriorityQueue) Len() int {
 }
 
 func (pq *PriorityQueue) Release(value Value) {
-	queue := &internalQueue{}
-	heap.Init(queue)
+	idx := pq.findIdx(func(item *Item) bool {
+		return item.value == value
+	})
 
-	for _, item := range *pq.queue {
-		if item.value != value {
-			heap.Push(queue, item)
-		}
+	if idx == -1 {
+		return
+	}
+
+	lastItem := pq.queue.Pop().(*Item)
+
+	if (*pq.queue).Len() == 0 || idx == (*pq.queue).Len() {
+		return
 	}
 
-	pq.queue = queue
+	(*pq.queue)[idx] = lastItem
+	heap.Fix(pq.queue, idx)
```

Comment:
When I first implemented this, I didn't look at the `heap` package properly. Through this PR, I learned that the `heap` already has an element deletion function. Thank you. 👍 

How about using `heap.Remove` directly after finding the index of the element?
```suggestion
	heap.Remove(pq.queue, idx)
```